### PR TITLE
Adds Primary field to Department-Thesis relationship

### PR DESCRIPTION
**Pull request by matt-bernhardt**
This adds the Primary flag to the Department_Thesis table. There are some associated changes with this ticket which are described in the commit message. I'd rather not type them all out here again, but will copy-paste if reviewers want.

Ticket: https://mitlibraries.atlassian.net/browse/ETD-64

#### Developer

- [x] All new ENV is documented in README
- [x] All new ENV has been added to Heroku Pipeline, Staging and Prod
- [x] ANDI or Wave has been run in accordance to
      [our guide](https://mitlibraries.github.io/guides/basics/a11y.html) and
      all issues introduced by these changes have been resolved or opened as new
      issues (link to those issues in the Pull Request details above)
- [x] Stakeholder approval has been confirmed (or is not needed)

#### Code Reviewer

- [x] The commit message is clear and follows our guidelines
      (not just this pull request message)
- [x] There are appropriate tests covering any new functionality
- [x] The documentation has been updated or is unnecessary
- [x] The changes have been verified
- [x] New dependencies are appropriate or there were no changes

#### Requires database migrations?

YES

#### Includes new or updated dependencies?

NO

### CODE REVIEW

File: test/integration/admin_dashboard_test.rb [hunk 1]
```diff
@@ -388,6 +388,45 @@ def teardown
...
+
+  test 'accessing department_thesis panel as a thesis_admin user works' do
+    mock_auth(users(:thesis_admin))
+    get '/admin/department_theses'
+    assert_response :success
+    assert_equal('/admin/department_theses', path)
+  end
+
+  test 'can edit department_thesis through admin dashboard' do
+    needle = false
```
Comment (JPrevost) [inline test/integration/admin_dashboard_test.rb:416]: This definitely works, so can stay. However, I wanted to share the needle confused me when reading this and I'm not yet convinced (but am happy to be swayed) that it's clearer with the needle variable than just using `true` and `false` directly in the code. Looking at `!needle` and then having to look through to follow the state of `needle` to ensure it didn't change kinda broke my Friday afternoon brain lol

File: test/models/department_thesis_test.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  test 'valid link with and without primary' do
+    link = department_theses(:primary)
+    assert_equal link.primary, true
+    assert(link.valid?)
+    link = department_theses(:other)
+    assert_equal link.primary, false
+    assert(link.valid?)
+  end
+
+  test 'only one primary department per thesis' do
```
Comment (JPrevost) [inline test/models/department_thesis_test.rb:23]: Looking at this test doesn't make me confident without looking at the underlying data that the test does what it says. Would you consider making it clearer in the test what is going on?

**Review by JPrevost — CHANGES_REQUESTED** [review]
My code comments are non blocking.

However, when testing this in the PR build if I look at Department Theses in administrate I get an error if I have submitted a Thesis via the Student submission form.

This may be expected due to that form not collecting data in a way that works with this data model change? I'm not sure so thought we should look into that before proceeding.

File: test/integration/admin_dashboard_test.rb (hunk 1 above)
Comment (matt-bernhardt) [reply test/integration/admin_dashboard_test.rb:416]: Yeah, I can hear that feedback. I struggled a bit with whether I should use a variable for this. We use `needle` as a variable in tests several times throughout this file, but in this case it felt a bit awkward. Seeing your feedback on this makes me think that we should try it another way. Stay tuned...

File: test/models/department_thesis_test.rb (hunk 2 above)
Comment (matt-bernhardt) [reply test/models/department_thesis_test.rb:23]: Sure - looking at this again with fresh coffee, I can see that this is a bit sparse. I've got a longer test now that does more to explicitly test the underlying conditions prior to these actions.

File: test/integration/admin_dashboard_test.rb (hunk 1 above)
Comment (JPrevost) [reply test/integration/admin_dashboard_test.rb:416]: Yeah. To be clear I'm not opposed to a variable (named `needle` or not) in general. Just for booleans it was making me think harder than I wanted to lol

**Comment by matt-bernhardt** [comment]
The problems in the review app were apparently brought about by my squashing some earlier migrations down to one for the purposes of easier code review. Deleting the review app and re-creating it entirely caused the `id` field in the department_thesis table to be created as expected.

I've updated the tests to address the two comments that you made on Friday - that was good feedback, thanks.

**Review by JPrevost — APPROVED** [review]

**Comment by matt-bernhardt** [comment]
Helen has approved this change against the data model - going to merge now...

**Pull request merged**